src/components/comparison/layers-panel/layers-panel.tsx

```diff
@@ -161,7 +173,48 @@ export const LayersPanel = ({
     };
 
     onLayerInsert(newLayer);
-    setShowInsertPanel(false);
+    setShowLayerInsertPanel(false);
+  };
+
+  const handleInsertScene = async (scene: {
+    name: string;
+    url: string;
+    token?: string;
+  }) => {
+    const existedScene = layers.some(
+      (exisLayer) => exisLayer.url.trim() === scene.url.trim()
+    );
+
+    if (existedScene) {
+      setShowSceneInsertPanel(false);
+      setShowExistedSceneWarning(true);
+      return;
+    }
+
+    const webScene = await load(scene.url, ArcGisWebSceneLoader);
+
+    const children = webScene.layers.map((child) => ({
+      id: child.id,
+      name: child.title,
+      url: child.url,
+    }));
+
+    if (!children.length) {
+      setShowSceneInsertPanel(false);
+      setShowNoSupportedLayersInSceneWarning(true);
```

Comment:
That was about another logic. If the scene has at least one unsupported layer, the warning should be shown.
If the webscene doesn't contain the data type of children, we need to load layer.json of each child to check.